experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -0,0 +1,654 @@
+        R.rep_mat = M
+        R.sym_deg = (false, 0)
+        R.reynolds_v = reynolds_v_slm
+        return R
+    end
+end
+
+function representation_reductive_group(G::ReductiveGroup, d::Int)
+    return RepresentationReductiveGroup(G, d)
+end
+
+representation_matrix(R::RepresentationReductiveGroup) = R.rep_mat
+reductive_group(R::RepresentationReductiveGroup) = R.group
+function vector_space_dimension(R::RepresentationReductiveGroup)
+    return ncols(R.rep_mat)
+end
+
+function Base.show(io::IO, R::RepresentationReductiveGroup)
+    println(io, "Representation of ", R.group.group[1], R.group.group[2])
+    if R.sym_deg[1]
+        print(io, "over symmetric forms of degree ", R.sym_deg[2])
+    else
+        println(io, "with representation matrix")
+        show(io, R.rep_mat)
+    end
+end
+
+###############
+
+#for the representation matrices
+
+function rep_mat_(G::ReductiveGroup, sym_deg::Int, F::Field)
+    G.group[1] == :SL || error("Only implemented for SLm")
+    m = G.group[2]
+    n = binomial(m + sym_deg - 1, m - 1)
+    mixed_ring, t, z, a = PolynomialRing(F, "t"=> 1:m, "z"=> (1:m, 1:m), "a" => 1:n)
+    group_mat = matrix(mixed_ring, m,m,[z[i,j] for i in 1:m, j in 1:m])
+    vars = [t[i] for i in 1:m]
+    new_vars = group_mat*vars
+
+    degree_basiss = reverse(degree_basis(mixed_ring,m, sym_deg))
+    
+    # transform the degree_basiss elements
+    new_vars_plus_rest = vcat(new_vars, gens(mixed_ring)[m+1:nvars(mixed_ring)])
+    images_of_degree_basiss = [evaluate(f, new_vars_plus_rest) for f in degree_basiss]
+
+    mat = zero_matrix(mixed_ring, n, n)
+    for i in 1:n
+        f = images_of_degree_basiss[i]
+        # express f as a linear combination of degree_basis
+        for t in terms(f)
+            for j in 1:n
+                if divides(t, degree_basiss[j])[1]
+                    c = leading_coefficient(degree_basiss[j])
+                    mat[i,j] += t / degree_basiss[j] * c
+                    #mat[i,j] += t / degree_basiss[j]  # FIXME: this is the natural thing but gives "wrong" (?) results
+                end
+            end
+        end
+    end
+    #we have to return mat in a different ring! 
+    group_ring = base_ring(G.group_ideal)
+    mapp = hom(mixed_ring, group_ring, vcat([0 for i in 1:m], gens(group_ring), [0 for i in 1:n]))
+    Mat = matrix(group_ring, n, n, [mapp(mat[i,j]) for i in 1:n, j in 1:n])
+    return Mat            
+end
+
+
+function degree_basis(R::MPolyRing,m::Int, t::Int)
+    v = R(1)
+    genss = gens(R)
+    n = length(gens(R)[1:m])
+    C = zero_matrix(Int,n,n)
+    for i in 1:n
+      C[i,i] = -1 #find a better way to write this TODO
+    end
+    d = [0 for i in 1:n]
+    A = [1 for i in 1:n]
+    b = [t]
+    P = Polyhedron((C,d),(A,b))
+    L = lattice_points(P)
+    W = Vector{MPolyRingElem}(undef,0)
+    for l in L
+        for i in 1:n
+            v = v*genss[i]^l[i]
+        end
+        v = v*multinomial(t,l)
+        push!(W,v)
+        v = R(1)
+    end
```

Comment:
More efficient: construct a polynomial from a list of coefficients and a list of exponent vectors (here each vector consists of just one entry, of course)
```suggestion
    W = [ R([multinomial(t,l)], [l]) for l in L ]
```

Reply:
Here I am getting a method error:
MethodError: no method matching (::QQMPolyRing)(::Vector{Int64}, ::Vector{PointVector{ZZRingElem}})
Closest candidates are:
  (::MPolyRing)(::Vector, ::Array{Vector{T}, 1}) where T at ~/.julia/packages/AbstractAlgebra/YkCOC/src/MPoly.jl:1279